From bc3f3054959ffb3a47aaea0736e3e10eeb9428b3 Mon Sep 17 00:00:00 2001
From: David Daney <ddaney@caviumnetworks.com>
Date: Fri, 14 May 2010 14:22:55 -0700
Subject: [PATCH 219/337] MIPS: Octeon: Increase PCI_CONFIG_SPACE_DELAY

Based On SDK 3.0.0-482

Some PCI cards require delays when accessing config space. A Sil3124
behind an Intel bridge will fail with 10ms and succeed with 11ms. The
setting below of 15ms adds some extra.

Signed-off-by: David Daney <ddaney@caviumnetworks.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
Signed-off-by: Jack Tan <jack.tan@windriver.com>
---
 arch/mips/include/asm/octeon/pci-octeon.h |    8 ++++++--
 1 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/mips/include/asm/octeon/pci-octeon.h b/arch/mips/include/asm/octeon/pci-octeon.h
index c66734b..07ab6a0 100644
--- a/arch/mips/include/asm/octeon/pci-octeon.h
+++ b/arch/mips/include/asm/octeon/pci-octeon.h
@@ -11,8 +11,12 @@
 
 #include <linux/pci.h>
 
-/* Some PCI cards require delays when accessing config space. */
-#define PCI_CONFIG_SPACE_DELAY 10000
+/*
+ * Some PCI cards require delays when accessing config space. A
+ * Sil3124 behind an Intel bridge will fail with 10ms and succeed with
+ * 11ms. The setting below of 15ms adds some extra.
+ */
+#define PCI_CONFIG_SPACE_DELAY 15000
 
 /*
  * The physical memory base mapped by BAR1.  256MB at the end of the
-- 
1.7.5.4

